src/main/resources/org/jboss/xavier/analytics/rules/workload/inventory/ReasonableDefaults.drl

```diff
@@ -41,3 +41,14 @@ rule "Fill 'Insights' field with reasonable default"
     then
         workloadInventoryReport.setInsightsEnabled(WorkloadInventoryReportModel.INSIGHTS_ENABLED_DEFAULT_VALUE);
 end
+
+rule "Fill 'osName' field with reasonable default"
+       when
+           workloadInventoryReport : WorkloadInventoryReportModel(
+               (osName == null && osDescription == null) ||
+               (osName != null && osName.isEmpty() && osDescription != null && osDescription.isEmpty())
+           )
+       then
+           workloadInventoryReport.setOsName(WorkloadInventoryReportModel.OS_NAME_DEFAULT_VALUE);
+           workloadInventoryReport.setOsDescription(WorkloadInventoryReportModel.OS_NAME_DEFAULT_VALUE);
+   end
```

Comment:
In my understanding, rules are not fired in a particular order in our case. And there is the rule `Target_None` that is executing the method `isUnsupported` that is doing a multi AND condition ( therefore evaluating each condition ) with `isUndetectedOS` AND `isSupportedOS` that executes the method `OSSupport.findOSSupportForOS` that does a `.filter(os -> osName.toLowerCase().contains(os.getName().toLowerCase()))` which could result on a `NPE` in the case of `OSName == null,OSDescription="something"`.  
wdyt ?

Reply:
If the input OS could be null, isUnsupportedOS would then evaluate as false on its call into !isUndetectedOS ( the OS being null  would cause isUndetectedOS == true on its first check there). My intention was that the operators in these methods would short-circuit, are you saying they won't?

 Anyway though, we always pass osDescription into the HelperFunctions methods and this PR ensures osDescription can't be null because ReasonableDefaults fires before Targets rules do. 

So osname == null doesn't matter in terms of evaluation of other rules. Maybe it does for ui display purposes, I don't know.

Reply:
:+1: